tests/e2e/upgrade_suite_test.go

```diff
@@ -0,0 +1,186 @@
+package e2e_test
+
+import (
+	"context"
+	"log"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	"github.com/operator-framework/api/pkg/operators/v1"
+	"github.com/operator-framework/api/pkg/operators/v1alpha1"
+	velerov1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	oadpv1alpha1 "github.com/openshift/oadp-operator/api/v1alpha1"
+	"github.com/openshift/oadp-operator/tests/e2e/lib"
+)
+
+type channelUpgradeCase struct {
+	previous   string
+	next       string
+	production bool
+}
+
+var _ = ginkgo.Describe("OADP upgrade scenarios", ginkgo.Ordered, func() {
+	ginkgo.DescribeTable("Upgrade OADP channel tests",
+		func(scenario channelUpgradeCase) {
+			// Create OperatorGroup and Subscription with previous channel stable-1.4
+			log.Print("Checking if OperatorGroup needs to be created")
+			operatorGroupList := v1.OperatorGroupList{}
+			err := runTimeClientForSuiteRun.List(context.Background(), &operatorGroupList, client.InNamespace(namespace))
+			gomega.Expect(err).To(gomega.BeNil())
+			gomega.Expect(len(operatorGroupList.Items) > 1).To(gomega.BeFalse())
+			if len(operatorGroupList.Items) == 0 {
+				log.Print("Creating OperatorGroup oadp-operator-group")
+				operatorGroup := v1.OperatorGroup{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "oadp-operator-group",
+						Namespace: namespace,
+					},
+					Spec: v1.OperatorGroupSpec{
+						TargetNamespaces: []string{namespace},
+					},
+				}
+				err = runTimeClientForSuiteRun.Create(context.Background(), &operatorGroup)
+				gomega.Expect(err).To(gomega.BeNil())
+			}
+
+			subscriptionPackage := "oadp-operator"
+			subscriptionSource := "oadp-operator-catalog-test-upgrade"
+			if scenario.production {
```

Comment:
ah this is cool, that you have enabled this to test dev or production builds.  Nice